rbx_binary/src/serializer/state.rs

```diff
@@ -580,6 +611,16 @@ impl<'dom, W: Write> SerializerState<'dom, W> {
                         // reasonable default.
                         Cow::Borrowed(prop_info.default_value.borrow())
                     })
+                    .map(|value| {
+                        if let Some(migration) = prop_info.migration {
+                            match migration.perform(&value) {
+                                Ok(new_value) => Cow::Owned(new_value),
+                                Err(_) => value,
+                            }
+                        } else {
+                            value
+                        }
```

Comment:
Migrations seem unlikely to fail anyway, so I think this is fine. The one we might run into is someone setting an invalid `Font` enum value, but I don't think that's something we can reasonably account for right now.

Reply:
I think this might be a little stranger than you're realizing - the Err branch will be hit under normal operation. The reason why is because in this PR, all migrated properties are collated under a single `PropInfo`. For reference, here's what `PropInfo` looks like during the migrated properties test:
```
PropInfo {
    prop_type: Color3uint8,
    serialized_name: "Color3uint8",
    aliases: {
        "BrickColor",
        "brickColor",
    },
    default_value: Color3uint8(
        Color3uint8 {
            r: 163,
            g: 162,
            b: 165,
        },
    ),
    migration: Some(
        PropertyMigration {
            new_property_name: "Color",
            migration: BrickColorToColor,
        },
    ),
}
```
This means that in `serialize_properties`, some properties will be `Variant::BrickColor`, some properties will be `Variant::Color3uint8` or `Variant::Color3`, and the migration will fail - but we expect it to!

Reply:
Oh that's kinda gross. I wonder if it's worth attempting a conversion in the same way we do with rbx_xml or if it's just a lost cause?

Reply:
Since this is likely throwaway code I'm not sure if it matters too much - any mismatches will be caught later in the `match prop_info.prop_type`.  I tried to make this less ugly, but I wasn't able to do so while still having one prop chunk per logical property. I don't think there's a good way to do it without significantly changing the binary serializer

Reply:
I suppose another option is we make the migration catch multiple types of variants? That feels less technically correct but is probably safer anyway.

Reply:
I don't think that's necessary. Besides applying migrations for select properties, there's no other changes to the behavior of `serialize_properties` as a result of this, and I'll want to delete this code ASAP regardless

Reply:
That's fair enough. We're probably good to merge this then.